crates/rpc/rpc-engine-api/src/engine_api.rs

```diff
@@ -208,6 +208,32 @@ where
         &self,
         payload_id: PayloadId,
     ) -> EngineApiResult<ExecutionPayloadEnvelopeV3> {
+        // First get the best payload, do not resolve it (we need to check the timestamp). If we
+        // were to resolve the payload here, and it is before cancun, then it's possible a
+        // subsequent call to `get_payload_v3` would return `UnknownPayload` because the payload
+        // may no longer exist in the payload store.
```

Comment:
I don't understand this,
The timestamp is fixed, why would we expect a subsequent call to get payload? 

it's a bit weird that we're effectively calling a getter twice.

Reply:
We typically wouldn't expect another call, but this is how one of the hive tests works unfortunately

Reply:
Just added another method to the payloadstore that returns the payloadattributes